src/main/webapp/app/exam/exam-scores/exam-scores.component.ts

```diff
@@ -387,6 +396,24 @@ export class ExamScoresComponent implements OnInit, OnDestroy {
                 examStatistics.meanGradePassed = this.gradingSystemService.findMatchingGradeStep(this.gradingScale!.gradeSteps, examStatistics.meanPointsRelativePassed)!.gradeName;
                 examStatistics.medianGradePassed = this.gradingSystemService.findMatchingGradeStep(this.gradingScale!.gradeSteps, examStatistics.medianRelativePassed)!.gradeName;
             }
+            // Calculate statistics for the first assessments of passed exams if second correction exists
+            if (this.hasSecondCorrectionAndStarted) {
+                examStatistics.meanPointsPassedInFirstCorrection = SimpleStatistics.mean(studentPointsPassedInFirstCorrectionRound);
+                examStatistics.medianPassedInFirstCorrection = SimpleStatistics.median(studentPointsPassedInFirstCorrectionRound);
+                examStatistics.standardDeviationPassedInFirstCorrection = SimpleStatistics.standardDeviation(studentPointsPassedInFirstCorrectionRound);
+                if (this.examScoreDTO.maxPoints) {
+                    examStatistics.meanPointsRelativePassedInFirstCorrection = (examStatistics.meanPointsPassedInFirstCorrection / this.examScoreDTO.maxPoints) * 100;
+                    examStatistics.medianRelativePassedInFirstCorrection = (examStatistics.medianPassedInFirstCorrection / this.examScoreDTO.maxPoints) * 100;
+                    examStatistics.meanGradePassedInFirstCorrection = this.gradingSystemService.findMatchingGradeStep(
+                        this.gradingScale!.gradeSteps,
+                        examStatistics.meanPointsRelativePassedInFirstCorrection,
+                    )!.gradeName;
+                    examStatistics.medianGradePassedInFirstCorrection = this.gradingSystemService.findMatchingGradeStep(
+                        this.gradingScale!.gradeSteps,
+                        examStatistics.medianRelativePassedInFirstCorrection,
+                    )!.gradeName;
+                }
+            }
```

Comment:
This gets repeated 3 times so I think you should put it into a function and handle small differences with arguments

Reply:
In general I think this is a very huge function which could be split up into more functions, not even only the new parts

Reply:
Well technically it does not get repeated 3 times. You access different attributes and use different values to calculate the mean/median etc.
I don't see how i can unify those actions by using parameters, but i can split it up in smaller functions if you like

Reply:
simplified in 482f9c5 :)